rust-runtime/aws-smithy-runtime-api/src/client/connection.rs

```diff
@@ -28,11 +29,13 @@ impl ConnectionMetadata {
     pub fn new(
         is_proxied: bool,
         remote_addr: Option<SocketAddr>,
+        local_addr: Option<SocketAddr>,
```

Comment:
Adding this argument is a breaking change in one of our stable crates, unfortunately. What you could do is write a builder for this struct, and then deprecate this `new` method in favor of it.

Reply:
Thanks for pointing that out! I'll make another revision

Reply:
I force pushed a new revision that has a builder type, reverted the `local_addr` parameter from the `new` method, and deprecated the `new` method.